Add --{enable,disable}-nosem option

dinosaure opened the pull request:
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

This PR adds options: `--{disable,enable}-nosem` as we can found the, into the Python code.

dinosaure committed:
Add --{enable,disable}-nosem option

aryx commented on src/osemgrep/cli_scan/Scan_CLI.ml:
```diff
@@ -604,6 +606,16 @@ let o_project_root : string option Term.t =
+          a 'nosem' comment at the end.|}
+  in
+  let disable = Arg.info [ "disable-nosem" ] in
+  Arg.value (Arg.vflag true [ (true, enable); (false, disable) ])
```
there is an H.negatable_flag function that we usually use for this.

aryx commented on src/osemgrep/cli_scan/Scan_CLI.ml:
```diff
@@ -617,7 +629,7 @@ let cmdline_term : conf Term.t =
       scan_unknown_extensions severity show_supported_languages strict
       target_roots test test_ignore_todo time_flag timeout timeout_threshold
-      validate verbose version version_check vim =
+      validate verbose version version_check vim nosem =
```
please add entries in alphabetic order. See the comment above.

dinosaure committed:
Apply @aryx's review about nosem option

- respect the lexicographic order of options
- Use `negatable_flag` instead of basics Cmdliner functions

aryx commented on src/osemgrep/cli_scan/Scan_CLI.ml:
```diff
@@ -613,11 +621,11 @@ let cmdline_term : conf Term.t =
-      scan_unknown_extensions severity show_supported_languages strict
-      target_roots test test_ignore_todo time_flag timeout timeout_threshold
-      validate verbose version version_check vim =
+      max_memory_mb max_target_bytes metrics num_jobs nosem optimizations
```
o is before u in the alphabet usually :)

mjambon replied:
@aryx could semgrep be used to enforce the alphabetic ordering of lists?

I posted a feature suggestion: https://github.com/returntocorp/semgrep/issues/7457
